app/src/main/java/nerd/tuxmobil/fahrplan/congress/settings/SettingsFragment.kt

```diff
@@ -80,6 +100,41 @@ class SettingsFragment : PreferenceFragment() {
         }
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (REQUEST_CODE_ALARM_TONE == requestCode && data != null) {
+            val alarmToneUri = data.getParcelableExtra<Uri>(RingtoneManager.EXTRA_RINGTONE_PICKED_URI)
+            onAlarmTonePicked(alarmToneUri)
+        } else {
+            super.onActivityResult(requestCode, resultCode, data)
+        }
+    }
+
+    private fun onAlarmTonePicked(alarmToneUri: Uri?) {
+        appRepository.updateAlarmToneUri(alarmToneUri)
+    }
+
+    override fun onPreferenceTreeClick(preference: Preference?): Boolean {
+        return if (getString(R.string.preference_key_alarm_tone) == preference?.key) {
```

Comment:
Just curious: any particular reason you wrote the comparison in this order and not the other way around?

Reply:
I prefer to put the value which I "control" (not `null`) to the left side of the comparison, here this is `R.string.preference_key_alarm_tone`. The `preference` parameter is nothing I "control".

Reply:
:see_no_evil:  We can't be friends anymore :sob: